lib/matplotlib/text.py

```diff
@@ -140,7 +135,12 @@ def _text_metrics(text, fontprop, ismath, dpi):
                     "This should never happen and is evidence of a bug elsewhere."
                     )
             # do the actual method call we need and return the result
-            return local_renderer.get_text_width_height_descent(text, fontprop, ismath)
+            # We make a copy of the fontprop because FontProperties is both mutable and
+            # has a `__hash__` that depends on that mutable state.  This is not ideal
+            # as it means the hash of an object is not stable over time which leads to
+            # very confusing behavior when used as keys in dictionaries or hashes.
+            return local_renderer.get_text_width_height_descent(text,
+                                                                fontprop.copy(), ismath)
```

Comment:
I do not understand the copy here. AFAIK `get_text_width_height_descent()` only reads from fontprop, so there should not be a benefit of passing a copy.